### PR TITLE
Add script to test notebooks

### PR DESCRIPTION
**Pull request by sandcha**
Connected to #37 

* In order to check notebooks statuses at OpenFisca evolutions, this starts by scripting notebooks checks for a specific version of OpenFisca (aka `openfisca-france` here).
* Details: 
  * Adds `test_notebooks.py` script to automatically check existing notebooks execution
  * Adds a list of all notebooks dependencies in `requirements.txt`
  * Adds a `Makefile` to allow easy install, execution and cleaning.

### CODE REVIEW

File: notebooks/test_notebooks.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
+# -*- coding: utf-8 -*-
+
+import sys
+import os
+from io import open
+
+from nbformat import read, write
+from nbconvert.preprocessors import ExecutePreprocessor, CellExecutionError
+# Other dependencies: jupyter-client, ipykernel
```
Comment (maukoquiroga) [inline notebooks/test_notebooks.py]: What's the meaning of this comment? Could we remove it?

File: notebooks/test_notebooks.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+import logging
+import traceback
+
+
+'''
+Checks notebooks by trying to convert each one into a notebook
+that includes its execution results.
+
+This script is similar to this command:
+jupyter nbconvert --to notebook --execute --ExecutePreprocessor.timeout=60
```
Comment (maukoquiroga) [inline notebooks/test_notebooks.py]: What's then the idea of this script ?

If we can execute :

```
jupyter nbconvert --to notebook --execute --ExecutePreprocessor.timeout=60 --output executed_notebook.ipynotebook demo.ipynb
```

We'll know that if it fails, then for example CircleCI should fail

File: notebooks/Makefile [hunk 3]
```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
```
Comment (maukoquiroga) [inline notebooks/Makefile]: Est-ce que cela a un effet quelconque ?

File: notebooks/Makefile [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+
+all: test
+
+install:
+	pip install -r requirements.txt
+
+test: clean
+	python test_notebooks.py .
+
+clean:
```
Comment (maukoquiroga) [inline notebooks/Makefile]: J'ai un erreur d'execution :

```
hyperion@diotimac:~/Sites/dinsic/openfisca/tutorial/notebooks[test-notebooks-37] > make
rm executed_*
rm: make: *** [clean] Abort trap: 6 (core dumped)
```

Comment (maukoquiroga) [reply notebooks/Makefile]:
```
hyperion@diotimac:~/Sites/dinsic/openfisca/tutorial/notebooks[test-notebooks-37] > make
/bin/rm executed_*
rm: executed_*: No such file or directory
make: *** [clean] Error 1
```

**Comment by maukoquiroga** [comment]
I'm fixing Makefile in local to see if the test runs... and it works! Here's the output:

```
hyperion@diotimac:~/Sites/dinsic/openfisca/tutorial/notebooks[test-notebooks-37*] > make 
rm -f executed_*
python test_notebooks.py .
> how_to_handle_periods.ipynb
Error executing the notebook "how_to_handle_periods.ipynb".

See notebook "./executed_how_to_handle_periods.ipynb" for traceback.
CellExecutionError: 
<bound method type.from_cell_and_msg of <class 'nbconvert.preprocessors.execute.CellExecutionError'>>
Traceback (most recent call last):
  File "test_notebooks.py", line 80, in <module>
    run(os.path.join(target, file))
  File "test_notebooks.py", line 54, in run
    {'metadata': {'path': notebook_directory}}
  File "/Users/hyperion/.pyenv/versions/tutoriel/lib/python2.7/site-packages/nbconvert/preprocessors/execute.py", line 361, in preprocess
    nb, resources = super(ExecutePreprocessor, self).preprocess(nb, resources)
  File "/Users/hyperion/.pyenv/versions/tutoriel/lib/python2.7/site-packages/nbconvert/preprocessors/base.py", line 69, in preprocess
    nb.cells[index], resources = self.preprocess_cell(cell, resources, index)
  File "/Users/hyperion/.pyenv/versions/tutoriel/lib/python2.7/site-packages/nbconvert/preprocessors/execute.py", line 385, in preprocess_cell
    raise CellExecutionError.from_cell_and_msg(cell, out)
CellExecutionError: An error occurred while executing the following cell:
------------------
simulation.calculate('af', '2015')
------------------

---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-2-6c6f6a2075ab> in <module>()
----> 1 simulation.calculate('af', '2015')

/Users/hyperion/.pyenv/versions/tutoriel/lib/python2.7/site-packages/openfisca_core/simulations.pyc in calculate(self, variable_name, period, **parameters)
    153             self.tracer.record_calculation_start(variable.name, period, **parameters)
    154 
--> 155         self._check_period_consistency(period, variable)
    156 
    157         extra_params = parameters.get('extra_params', ())

/Users/hyperion/.pyenv/versions/tutoriel/lib/python2.7/site-packages/openfisca_core/simulations.pyc in _check_period_consistency(self, period, variable)
    288                 variable.name,
    289                 period
--> 290                 ).encode('utf-8'))
    291 
    292         if variable.definition_period == periods.YEAR and period.unit != periods.YEAR:

ValueError: Unable to compute variable 'af' for period 2015: 'af' must be computed for a whole month. You can use the ADD option to sum 'af' over the requested period, or change the requested period to 'period.first_month'.
ValueError: Unable to compute variable 'af' for period 2015: 'af' must be computed for a whole month. You can use the ADD option to sum 'af' over the requested period, or change the requested period to 'period.first_month'.
```

File: notebooks/test_notebooks.py (hunk 2 above)
Comment (adrienpacifico) [reply notebooks/test_notebooks.py]:
```
jupyter nbconvert --to notebook --execute --ExecutePreprocessor.timeout=60 --output executed_notebook.ipynotebook demo.ipynb
```

Why did you choose to limit timeout to 60 seconds ?

**Pull request merged**